### PR TITLE
Integrate planning tests

### PR DESCRIPTION
**Pull request by mjeronimo**
Integrate Carlos' planner tests.

### CODE REVIEW

File: src/planning/nav2_planning_tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+find_package(nav2_world_model_msgs REQUIRED)
+find_package(nav_msgs REQUIRED)
+find_package(visualization_msgs REQUIRED)
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_gtest REQUIRED)
+find_package(ament_cmake_pytest REQUIRED)
+
+set(TEST_LAUNCH_DIR ${CMAKE_CURRENT_SOURCE_DIR}/test_launch_files)
+
+ament_add_gtest_executable(test_planner_node
```
Comment (crdelsey) [inline src/planning/nav2_planning_tests/CMakeLists.txt]: All these cmake commands to build and add the tests should probably go in the `BUILD_TESTING` section. It means there won't be anything to build if that flag is not set, but I think that is correct behavior.

File: src/planning/nav2_planning_tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  GENERATE_RESULT_FOR_RETURN_CODE_ZERO
+  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/test_planner_launch.py"
+  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+  ENV
+    TEST_LAUNCH_DIR=${TEST_LAUNCH_DIR}
+    TEST_EXECUTABLE=$<TARGET_FILE:test_planner_node>
+    TEST_MAP=${CMAKE_CURRENT_SOURCE_DIR}/maps/map.pgm
+)
+
+install(TARGETS test_planner_node
```
Comment (crdelsey) [inline src/planning/nav2_planning_tests/CMakeLists.txt]: I don't think we should install the test executable.

File: src/planning/nav2_planning_tests/planner_tester.cpp [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+  if (!costmap_set_) {
+    RCLCPP_ERROR(
+      this->get_logger(),
+      "PlannerTester::startCostmapServer:: you need to set a costmap before starting the service");
+    return;
+  }
+
+  auto costmap_service_callback = [this](
+    const std::shared_ptr<rmw_request_id_t>/*request_header*/,
+    const std::shared_ptr<nav2_world_model_msgs::srv::GetCostmap::Request> request,
```
Comment (crdelsey) [inline src/planning/nav2_planning_tests/planner_tester.cpp]: We should start using typedefs and using directives. This is really awful to read.

File: src/planning/nav2_planning_tests/test_planner_node.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  auto result = std::make_shared<nav2_tasks::ComputePathToPoseResult>();
+
+  for (auto costmap : costmaps) {
+    loadSimpleCostmap(costmap);
+    EXPECT_EQ(true, defaultPlannerTest(result));
+  }
+}
+
+TEST_F(PlannerTester, testWithOnFixedEndpoints)
```
Comment (crdelsey) [inline src/planning/nav2_planning_tests/test_planner_node.cpp]: Should this test be called testWith*One*FixedEndpoint?

File: src/planning/nav2_planning_tests/CMakeLists.txt (hunk 1 above)
Comment (mjeronimo) [reply src/planning/nav2_planning_tests/CMakeLists.txt]: Fixed.

File: src/libs/nav2_util/src/costmap.cpp [hunk 5]
```diff
@@ -28,152 +29,240 @@ const Costmap::CostValue Costmap::medium_cost = 128;
...
-Costmap::Costmap(rclcpp::Node * node)
-: node_(node)
+Costmap::Costmap(
+  rclcpp::Node * node, bool trinary_costmap, bool track_unknown_space,
+  int lethal_threshold, int unknown_cost_value)
+: node_(node), trinary_costmap_(trinary_costmap), track_unknown_space_(track_unknown_space),
+  lethal_threshold_(lethal_threshold), unknown_cost_value_(unknown_cost_value)
 {
-  RCLCPP_INFO(node_->get_logger(), "Costmap::Costmap");
+  lethal_threshold_ = std::max(std::min(lethal_threshold_, 100), 0);
```
[the remark this answers is not in the document]
Comment (mjeronimo) [reply src/libs/nav2_util/src/costmap.cpp]: I'll enter an issue for Carlos for this item.

**Pull request merged**